client/src/store/utils/reducerFunctions.js

```diff
@@ -24,6 +24,19 @@ export const addMessageToStore = (state, payload) => {
   });
 };
 
+export const setReadMessageToStore = (state, payload) => {
+  const { otherUserId, conversationId } = payload;
+  return state.map((convo) => {
+    if (convo.id === conversationId && convo.otherUser.id === otherUserId) {
+      const convoCopy = { ...convo };
+      convoCopy.messages.forEach((message) => (message.isRead = true));
```

Comment:
I would suggest adding a small statement inside the loop to make sure you are only changing the `isRead` if it's not what you want. Otherwise you're going to re-assign it to `true` even in the case where `isRead` is already `true`.